### PR TITLE
Added point in time APIs

### PR DESCRIPTION
**Pull request by ajygupta**
### Description
Added point in time APIs

### Issues Resolved
_List any issues this PR will resolve, e.g. Closes [...]._ 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by Arpit-Bandejiya** [comment]
Should we add the documentation for using these APIs as well?

File: api/api/get_all_pits.js
Comment (bharath-techie) [inline api/api/get_all_pits.js:20]: https://github.com/opensearch-project/opensearch-js/pull/337/files --- Lets add JSDoc like in this PR to all the APIs.

**Comment by nhtruong** [comment]
> Should we add the documentation for using these APIs as well?

Yes please. And don't forget to update the changelog!
Thank you for your contribution :D

**Comment by ananzh** [comment]
Hey @ajygupta I see there are 6 old commits here, could you rebase it to fetch the latest?

File: api/api/create_pit.js
Comment (bharath-techie) [inline api/api/create_pit.js:41]: since the api supports multiple indices , why are we not using 'indices' ?

File: api/types.d.ts
Comment (bharath-techie) [inline api/types.d.ts:9979]: Lets call it indices ?

File: api/api/create_pit.js
Comment (ajygupta) [reply api/api/create_pit.js:41]: As discussed, to align with the convention used, we are going with `index`

File: USER_GUIDE.md
Comment (ananzh) [inline USER_GUIDE.md:229]: Could you add a comment here for how a user can get this `pit_id` ?

File: api/api/create_pit.js
Comment (ananzh) [inline api/api/create_pit.js:1]: I think the license header for new added file should be 

```
/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */
```

File: api/api/create_pit.js
Comment (ananzh) [inline api/api/create_pit.js:73]: nit: is `path` changeable? if not, should we change to `const`?

File: api/api/delete_all_pits.js
Comment (ananzh) [inline api/api/delete_all_pits.js:1]: same license issue

File: api/api/delete_pit.js
Comment (ananzh) [inline api/api/delete_pit.js:1]: same license issue

File: api/api/get_all_pits.js
Comment (ananzh) [inline api/api/get_all_pits.js:1]: same license issue

File: api/api/create_pit.js
Comment (ananzh) [inline api/api/create_pit.js:13]: What is the meaning of this two lines of eslint rules? could you provide some explains for adding these two lines? for example, is this `eslint camelcase: 0` same as `esline-disable camelcase` which disable camelcase check for a file?

File: api/api/create_pit.js
Comment (ananzh) [inline api/api/create_pit.js:41]: nit: I see other param has [] added like this one `@param {string} [params.routing]`, but line 41 and 42 don't have. Could you make them consistent?

File: api/api/delete_pit.js
Comment (ananzh) [inline api/api/delete_pit.js:2]: same license issue

File: api/api/get_all_pits.js
Comment (ananzh) [inline api/api/get_all_pits.js:2]: same license issue

**Review by ananzh — CHANGES_REQUESTED** [review]

File: api/api/create_pit.js
Comment (ajygupta) [reply api/api/create_pit.js:13]: This is just to keep it consistent with the other files, so that once we fix the auto-generation logic, we won't see any changes in these files.

File: api/api/create_pit.js
Comment (ajygupta) [reply api/api/create_pit.js:41]: So the square braces would be for the optional params

File: api/api/delete_all_pits.js
Comment (ajygupta) [reply api/api/delete_all_pits.js:22]: Yeah, so this would generate the JSDOC as implemented [here](https://github.com/opensearch-project/opensearch-js/pull/337)

File: api/api/create_pit.js
Comment (ajygupta) [reply api/api/create_pit.js:73]: This is to keep it consistent with the other files, so that once we fix the auto-generation logic, we won't see any changes in these files.

File: USER_GUIDE.md
Comment (ajygupta) [reply USER_GUIDE.md:229]: I'm not sure if I understand this completely. A particular point in time (PIT) is denoted by a `pit_id`. It's like a particular index is denoted by `index_name`

File: api/api/create_pit.js
Comment (ajygupta) [reply api/api/create_pit.js:1]: Updated. Thanks

Comment (ajygupta) [reply api/api/create_pit.js:1]: @ananzh I updated as per the suggestion but the license check fails. Do we need to fix that?

File: USER_GUIDE.md
Comment (ananzh) [reply USER_GUIDE.md:229]: How do we get a pit_id?

File: api/api/create_pit.js
Comment (ananzh) [reply api/api/create_pit.js:1]: @ajygupta 

I think currently opensearch-js is using this header [here](https://github.com/opensearch-project/opensearch-js/blob/871a6669c9153d8161b3bbbce8747b86f9e6f758/.ci/license/license-mc.txt). Pls double check.
```
/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 */
```

Well, we have updated this header to the one I mentioned in the comment. 
Some discussions as reference
https://github.com/opensearch-project/opensearch-build/pull/2666#issuecomment-1260139239
https://github.com/opensearch-project/alerting/issues/136#issuecomment-966657180

However, it is not updated in opensearch-js and the header check is still checking the old one. To unblock you, I would suggest to use the old one and I will submit an issue or PR to update the header.

File: api/api/create_pit.js
Comment (ananzh) [reply api/api/create_pit.js:8]: Well I think I don't understand these two lines. Do you understand why we use these two lines?

Comment (ajygupta) [reply api/api/create_pit.js:13]: AFAIK `eslint camelcase: 0` and `eslint no-unused-vars: 0` are for disabling camelcase check and unused variables check respectively.

File: USER_GUIDE.md
Comment (ajygupta) [reply USER_GUIDE.md:229]: We get it when we create a point in time.

**Review by ananzh — APPROVED** [review]
LGTM. Thank you.

**Comment by wbeckler** [comment]
@nhtruong can you take a look at this PR? It's been merged, but it wouldn't hurt to have another look since it was only reviewed by one maintainer.

**Comment by bharath-techie** [comment]
@kavilla can you please take a look and approve for us to release this change

**Comment by kavilla** [comment]
I would bump the version of this package to 2.2 for this feature

**Comment by kavilla** [comment]
Which version is this feature change compatible with? 

If it's only works on an unreleased version of OpenSearch I would update this file:

https://github.com/opensearch-project/opensearch-js/blob/main/COMPATIBILITY.md

Like bump the version of this package to reflect the changes in case folks pull down a patch update and try to use these APIs and not get a failure.

**Pull request merged**
